### PR TITLE
Improvements for regions

### PR DESCRIPTION
**Pull request by bwilczek**
* add support for `region_class` and `class_collection` parameters to `has_one` and `has_many`
* make regions nestable by replacing `@browser` with `region_element` wherever necessary

### CODE REVIEW

File: lib/watirsome/accessors.rb [hunk 1]
```diff
@@ -192,11 +193,11 @@ module InstanceMethods
       #
       def grab_elements(method, watir_args, custom_args)
         if custom_args.empty?
-          @browser.__send__(method, watir_args)
+          region_element.__send__(method, watir_args)
```
[the remark this answers is not in the document]
Comment (bwilczek) [reply lib/watirsome/accessors.rb]: There are also `browser` and `parent` references if `region_element` is not enough. Should suffice :)

File: lib/watirsome/regions.rb [hunk 2]
```diff
@@ -6,8 +6,8 @@ module Regions
     # @param [Symbol] region_name
     # @param [Block] block
     #
-    def has_one(region_name, &block)
-      define_region_accessor(region_name, &block)
+    def has_one(region_name, at: nil, region_class: nil, &block)
```
[the remark this answers is not in the document]
Comment (bwilczek) [reply lib/watirsome/regions.rb]: I see your point. My reasoning was that a single region lives `at` a one, precise point, while a collection of many region lives `within` another node. These words just felt right to me, however I'm open for changes here - I'm not strongly attached to this convention :)

File: Gemfile.lock [hunk 3]
```diff
@@ -0,0 +1,76 @@
+PATH
```
[the remark this answers is not in the document]
Comment (bwilczek) [reply Gemfile.lock]: It is recommended by the bundler workflow though:
https://bundler.io/v1.12/bundler_workflow.html

[the remark this answers is not in the document]
Comment (p0deje) [reply Gemfile.lock]: Lockfile should be committed when we develop app, but not when we develop gem. There is a good article explaining that - https://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/.

**Comment by p0deje** [comment]
You, sir, deserve a 🍰!

**Pull request merged**